lib/spack/spack/test/cmd/buildcache.py

```diff
@@ -848,3 +850,192 @@ def test_buildcache_prune_orphaned_manifest(tmp_path, mutable_database, mock_gnu
+        buildcache(*command_args)
+
+
+def test_buildcache_create_view_empty(
+    tmp_path, mutable_config, mutable_database, mutable_mock_env_path
+):
+    mirror_directory = str(tmp_path)
+    mirror("add", "--unsigned", "my-mirror", mirror_directory)
+
+    # Push a spec to the cache
+    mpileaks_specs = mutable_database.query_local("mpileaks")
+    buildcache("push", "--update-index", "my-mirror", mpileaks_specs[0].format("{/hash}"))
+
+    # Make sure the view doesn't exist yet
+    with pytest.raises(spack.binary_distribution.FetchIndexError):
+        hashes_in_view = read_specs_in_index(mirror_directory, "test_view")
+
+    # Write a minimal lockfile (this is not validated/read by an enviornment)
+    empty_lockfile = tmp_path / "emptylock" / "spack.lock"
+    empty_lockfile.parent.mkdir()
+    empty_lockfile.write_text(
+        '{"_meta": {"lockfile-version": 1}, "roots": [], "concrete_specs": {}}', encoding="utf-8"
+    )
+    # Create a view with now specs
+    command_args = [
+        "create-view",
+        "--force",
+        "--name",
+        "test_view",
+        "my-mirror",
+        str(empty_lockfile),
+    ]
+    out = buildcache(*command_args)
+    assert "No specs found for view, creating an empty index" in out
+    hashes_in_view = read_specs_in_index(mirror_directory, "test_view")
+    assert not hashes_in_view
+
+
+@pytest.mark.parametrize("sources", ("hash", "specfile", "env", "lockfile"))
+def test_buildcache_create_view(
+    tmp_path, mutable_config, mutable_database, mutable_mock_env_path, sources
+):
+    mirror_directory = str(tmp_path)
+    mirror("add", "--unsigned", "my-mirror", mirror_directory)
+
+    # Push all of the mpileaks packages to the cache
+    mpileaks_specs = mutable_database.query_local("mpileaks")
+    for s in mpileaks_specs:
+        buildcache("push", "--update-index", "my-mirror", s.format("{/hash}"))
+
+    # Grab all of the hashes for each mpileaks spec
+    mpileaks_0_hashes = [s.dag_hash() for s in mpileaks_specs[0].traverse()]
+    mpileaks_1_hashes = [s.dag_hash() for s in mpileaks_specs[1].traverse()]
+    mpileaks_2_hashes = [s.dag_hash() for s in mpileaks_specs[2].traverse()]
+
+    # Make sure the view doesn't exist yet
+    with pytest.raises(spack.binary_distribution.FetchIndexError):
+        hashes_in_view = read_specs_in_index(mirror_directory, "test_view")
+
+    # Create a view using a parametrized source that contains one of the mpileaks
+    with globals()[f"spec_{sources}"](mpileaks_specs[0], tmp_path / "source_stage") as source_args:
+        command_args = ["create-view", "--force", "--name", "test_view", "my-mirror"]
+        if source_args:
+            command_args.extend(source_args)
+        buildcache(*command_args)
+
+    hashes_in_view = read_specs_in_index(mirror_directory, "test_view")
+    for h in mpileaks_0_hashes:
+        assert h in hashes_in_view
+
+    # Test failure due to existing index
+
+    # Test append
```

Comment:
Seems like these should be separate tests rather than one super test. It will make it easier to debug failures in the future. Could nest them in a test class.

Reply:
I have them all in the same test to test the commands against different states of a view index to ensure the command is doing what I expect for the options. I could break it up into more though.

Reply:
I broke the test into three x ways to pass a spec

* `spack update-view` to create a view and then makes sure that `--force` is required overwrite an existing view.

* `spack update-view --append ...`. I found a bug writing this one where using `--append` when an index didn't exist it would error (fixed). So this test tests the `--append` to a non-existent view creates the view, and then appending to an existing view (created by the above command) successfully appends.

* `spack update-view --force ` can overwrite an existing view